### ISSUE TITLE
Nesting quotes inside shellouts inside quotes drops whitespace

### ISSUE THREAD

**Issue by nelsam** [earthly/earthly#2589]
# Summary:

In the shell, once you descend into a shellout via either backticks or dollar-parens, you can re-use quotes that were used at a higher level. For example, `echo "$(echo "foo      bar")"` will echo `foo      bar`.

In earthly, `RUN echo "$(echo "foo      bar")"` echoes `foo bar`.

# Detail:

The way that we parse `QuotedAtomPart` is causing the atoms that we get back to be `[ "$(echo ", foo, bar, ")" ]`.  In a perfect world, we would recognize the shellout even inside a `QuotedAtomPart` so that the nested quotes inside of the shellout are _not_ recognized as an end to the `QuotedAtomPart`.

# Steps:

Using the following `Earthfile`:
```Earthfile
VERSION 0.7
FROM alpine:3.15

foo:
    RUN echo "$(echo "foo    bar")"
```

- Run `earthly +foo`

## Expected:

`foo    bar`

## Actual:

`foo bar`

**Comment by nelsam** [comment]
Interestingly, these spaces are preserved in `ENV`:

```Earthfile
VERSION 0.7
FROM alpine:3.15

foo:
    ENV FOO="$(echo "foo    bar")"
    RUN echo "$FOO"
```
outputs `foo    bar`.  There may be other cases that work as expected.

**Comment by alexcb** [comment]
this has been fixed in v0.7.0-rc2
